File: week6/Week6_day3_ISSS_J/app/src/main/java/com/example/luisenriquez/week6_day3_isss_j/data/remote/RemoteServiceHelper.java
package com.example.luisenriquez.week6_day3_isss_j.data.remote;

import com.example.luisenriquez.week6_day3_isss_j.data.remote.models.ISSResponse;
import com.example.luisenriquez.week6_day3_isss_j.utils.Constants;

import io.reactivex.Single;
import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

public class RemoteServiceHelper
{
    public Single<ISSResponse> getISSResponse()
    {
        Retrofit MyRetroFit = getRetroFit(Constants.URLS.ISS_API_BASE_URL);
        RemoteService MyService = MyRetroFit.create(RemoteService.class);

        return MyService.getISSResponse("33.925990299999995",
                                        "-84.476051");
    }

    private Retrofit getRetroFit(String baseUrl)
    {
        HttpLoggingInterceptor MyInterceptor = new HttpLoggingInterceptor();

        MyInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);

        OkHttpClient MyClient = new OkHttpClient.Builder()
                                                .addInterceptor(MyInterceptor)
                                                .build();

        return new Retrofit.Builder()
                            .baseUrl(baseUrl)
                            .client(MyClient)
                            .addConverterFactory(GsonConverterFactory.create())
                            .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                            .build();
    }
}

File: week6/Week6Day3_ISS/app/src/main/java/com/example/luisenriquez/week6day3_iss/data/remote/RemoteServiceHelper.kt
package com.example.luisenriquez.week6day3_iss.data.remote.models

import android.provider.SyncStateContract
import com.jakewharton.retrofit2.adapter.kotlin.coroutines.experimental.CoroutineCallAdapterFactory
import com.yeah.ruisu.realvolvocodingchallenge.data.remote.models.ForecastDatum
import com.yeah.ruisu.realvolvocodingchallenge.utils.Constants
import kotlinx.coroutines.experimental.Deferred
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import org.joda.time.LocalDate
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

class RemoteServiceHelper
{
    fun getWeatherForecast(WOEID : String) : Deferred<List<ForecastDatum>>
    {
        val retrofit = getRetrofit(Constants.BASE_URL)
        val service = retrofit.create(RemoteService::class.java)
        val localDate = LocalDate().plusDays(1)

        return service.getWeatherForecast(WOEID, localDate.toString("YYYY/MM/dd"))
    }

    private fun getRetrofit (baseUrl : String) : Retrofit
    {
        val interceptor = HttpLoggingInterceptor()

        interceptor.level = HttpLoggingInterceptor.Level.BODY

        val okHttpClient = OkHttpClient.Builder()
                                        .addInterceptor(interceptor)
                                        .build()

        return Retrofit.Builder()
                        .baseUrl(baseUrl)
                        .client(okHttpClient)
                        .addConverterFactory(GsonConverterFactory.create())
                        .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                        .addCallAdapterFactory(CoroutineCallAdapterFactory())
                        .build()
    }


}